staging/src/k8s.io/component-base/logs/options_test.go

```diff
@@ -37,7 +37,7 @@ func TestFlags(t *testing.T) {
 	want := `      --experimental-logging-sanitization   [Experimental] When enabled prevents logging of fields tagged as sensitive (passwords, keys, tokens).
                                             Runtime log sanitization may introduce significant computation overhead and therefore should not be enabled in production.
       --logging-format string               Sets the log format. Permitted formats: "text".
-                                            Non-default formats don't honor these flags: --add_dir_header, --alsologtostderr, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size, --logtostderr, --one_output, --skip_headers, --skip_log_headers, --stderrthreshold, --vmodule, --log-flush-frequency.
+                                            Non-default formats don't honor these flags: --add_dir_header, --alsologtostderr, --log-flush-frequency, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size, --logtostderr, --one_output, --skip_headers, --skip_log_headers, --stderrthreshold, --vmodule.
```

Comment:
I don't know. It was overlooked when drafting [the KEP for klog deprecation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components), so there hasn't really been a discussion around it.

There will be a need to configure flushing when adding split output streams for JSON with buffered info messages. There also will be a need to flush those messages on program exit. Therefore the flag and the logs API make sense to me.

But I'm also fine with deprecating `--log-flush-frequency` now, together with the klog flags. That also makes sense because the periodic flushing could also be a per-backend configuration option.

Reply:
Is the savings of buffering (CPU?) really worth the complexity (flags, timers...) and memory ?

Reply:
I'm not familiar with this code, so I don't really know what "log-flush-frequency" is actually doing.
So let me clarify my main point.

What I would like to see eventually is that not every "info" log is blocking. This is the pattern we use internally (i.e. error logs are blocking execution, info logs are buffered). This is problematic for cases where a component would be able to operate faster but it's blocking on e.g. spammy logs because of slower disk.
We've seen that in the part for kube-apiserver in particular, but not only.

Reply:
@wojtek-t: I think that answered @thockin's question. He wanted to know whether we can remove this flag and always write without buffering, i.e. remove the need for flushing.

How this flushing is configured and implemented is up for debate, but we should have support for it.

Reply:
ok